Testscripts/Linux/perf_apache.sh

```diff
@@ -0,0 +1,255 @@
+	errMsg="Please add/provide value for nicName in constants.sh. nicName=eth0/bond0"
+	LogErr "${errMsg}"
+	SetTestStateAborted
+	exit 1
+fi
+
+localaddress=$(hostname -i)
+
+Run_SSHCommand()
+{
+	ips="$1"
+	cmd="$2"
+	IFS=',' read -r -a array <<< "$ips"
+	for ip in "${array[@]}"
+	do
+		LogMsg "Execute ${cmd} on ${ip}"
+		if [[ ${localaddress} = ${ip} ]]; then
+			bash -c "${cmd}"
+		else
+			ssh "${ip}" "${cmd}"
+		fi
+	done
+}
+
+# Install apache on client and server Machine
+LogMsg "Configuring client ${client}..."
+install_apache
+if [ $? -ne 0 ]; then
+	LogMsg "Error: apache installation failed in ${client}.."
+	SetTestStateAborted
+	exit 1
+fi
+
+LogMsg "Configuring server ${server}..."
+Run_SSHCommand "${server}" ". $UTIL_FILE && install_apache"
+if [ $? -ne 0 ]; then
+	LogMsg "Error: apache installation failed in ${server}.."
+	SetTestStateAborted
+	exit 1
+fi
+
+GetDistro
+case "$DISTRO_NAME" in
+	oracle|rhel|centos)
+		web_server="httpd"
+		document_path="/var/www/html/test.dat"
+		;;
+
+	ubuntu|debian)
+		web_server="apache2"
+		document_path="/var/www/html/test.dat"
+		;;
+
+	sles|sle_hpc)
+		web_server="apache2"
+		document_path="/srv/www/htdocs/test.dat"
+		;;
+	*)
+		LogErr "Unsupported distribution for apache test"
+		SetTestStateSkipped
+		exit 1
+esac
+
+mpstat_cmd="mpstat"
+dstat_cmd="dstat"
+sar_cmd="sar"
+vmstat_cmd="vmstat"
+
+testName="apache"
+log_folder="$(pwd)/${testName}-test-logs"
+max_concurrency_per_ab=4
+max_ab_instances=16
+testConcurrency=(1 2 4 8 16 32 64 128 256 512 1024)
+
+Run_ab()
+{
+	current_concurrency=$1
+
+	if [ ${current_concurrency} -le 2 ]
+	then
+		total_requests=50000
+	elif [ ${current_concurrency} -le 128 ]
+	then
+		total_requests=100000
+	else
+		total_requests=200000
+	fi
+
+	ab_instances=$(($current_concurrency / $max_concurrency_per_ab))
+	if [ ${ab_instances} -eq 0 ]
+	then
+		ab_instances=1
+	fi
+	if [ ${ab_instances} -gt ${max_ab_instances} ]
+	then
+		ab_instances=${max_ab_instances}
+	fi
+
+	total_request_per_ab=$(($total_requests / $ab_instances))
+	concurrency_per_ab=$(($current_concurrency / $ab_instances))
+	concurrency_left=${current_concurrency}
+	requests_left=${total_requests}
+	while [ ${concurrency_left} -gt ${max_concurrency_per_ab} ]; do
+		concurrency_left=$(($concurrency_left - $concurrency_per_ab))
+		requests_left=$(($requests_left - $total_request_per_ab))
+		LogMsg "Running parallel ab command for: ${total_request_per_ab} X ${concurrency_per_ab}"
+		ab -n ${total_request_per_ab} -r -c ${concurrency_per_ab} http://${server}/test.dat >> ${log_folder}/${current_concurrency}.${testName}.bench.log & pid=$!
+		PID_LIST+=" $pid"
+	done
+
+	if [ ${concurrency_left} -gt 0 ]
+	then
+		LogMsg "Running parallel ab command left for: ${requests_left} X ${concurrency_left}"
+		ab -n ${requests_left} -r -c ${concurrency_left} http://${server}/test.dat >> ${log_folder}/${current_concurrency}.${testName}.bench.log & pid=$!
+		PID_LIST+=" $pid";
+	fi
+	trap "kill ${PID_LIST}" SIGINT
```

Comment:
Why SIGINT is used to terminate. Cant we have graceful exit ?

Reply:
This command is that if the process receives SIGINT signal, then "kill ${PID_LIST}" will be executed.